src/components/NewCommentForm.tsx

```diff
@@ -1,37 +1,166 @@
-import React from 'react';
+/* eslint-disable no-console */
+import React, { useCallback, useState } from 'react';
+import cn from 'classnames';
+import isEmail from 'validator/lib/isEmail';
+import { Comment } from '../types/Comment';
+import { postComment } from '../utils/api';
+import { Post } from '../types/Post';
+
+interface Props {
+  setComments: React.Dispatch<React.SetStateAction<Comment[] | null>>;
+  selectedPost: Post | null;
+}
+
+enum InputType {
+  NAME = 'name',
+  EMAIL = 'email',
+  BODY = 'body',
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  setComments,
+  selectedPost,
+}) => {
+  const [name, setName] = useState<string>('');
+  const [email, setEmail] = useState<string>('');
+  const [body, setBody] = useState<string>('');
+  const [isCommentAddLoading,
+    setIsCommentAddLoading] = useState<boolean>(false);
+
+  const [nameError, setNameError] = useState<boolean>(false);
+  const [emailError, setEmailError] = useState<boolean>(false);
+  const [bodyError, setBodyError] = useState<boolean>(false);
+
+  const handleInput = useCallback((event: React.ChangeEvent<HTMLInputElement
+  | HTMLTextAreaElement>,
+  type: InputType) => {
+    switch (type) {
+      case InputType.NAME: {
+        setName(event.target.value);
+        if (nameError) {
+          setNameError(false);
+        }
+
+        break;
+      }
+
+      case InputType.EMAIL: {
+        setEmail(event.target.value);
+        setEmailError(false);
+        break;
+      }
+
+      case InputType.BODY: {
+        setBody(event.target.value);
+        setBodyError(false);
+        break;
+      }
+
+      default: {
+        break;
+      }
+    }
+  }, [name, email, body]);
```

Comment:
There is no need to include 'name', 'email', and 'body' in the array of dependencies for the useCallback hook because they are not being used inside the hook.